src/SIM/FunctionSum.h

```diff
@@ -51,6 +51,10 @@ class FunctionSum : public FunctionBase
   //! \brief Returns a representative scalar equivalent of the function value.
   virtual double getScalarValue(const Vec3& X) const;
 
+protected:
+  //! \brief Constructor to allow empty initialization in subclasses.
+  FunctionSum() {}
+
```

Comment:
It's OK, but move L54-L57 before L28, such that both constructors are declared first (kinda IFEM standard)?

Reply:
done